src/Microsoft.VisualStudio.LanguageServices.Razor/VisualStudioRazorParser.cs

```diff
@@ -0,0 +1,151 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.VisualStudio.Text;
+using ITextBuffer = Microsoft.VisualStudio.Text.ITextBuffer;
+
+namespace Microsoft.VisualStudio.LanguageServices.Razor
+{
+    public class VisualStudioRazorParser : IDisposable
+    {
+        private BackgroundParser _parser;
+        private RazorSyntaxTreePartialParser _partialParser;
+        private ITextBuffer _textBuffer;
+        private RazorCodeDocument _currentDocument;
+        private SourceChange _pendingShimTextChange;
+
+        public VisualStudioRazorParser(RazorTemplateEngine templateEngine, string filePath)
```

Comment:
Has Todd look at this updated design? now would be a good time to get his feedback.